### PR TITLE
Switch fxa_client from using mockiato to using mockall 

### LINKED ISSUE #5368: Switch fxa_client from using mockiato to using mockall
mockiato is a trait mocking library for unit tests used throughout fxa_client. Mockiato hasn't been updated in 4 or so years, and [mockall](https://crates.io/crates/mockall) seems like what the Rust community is adopting for trait mocking.

This ticket will be a great start for someone new to fxa_client to get around the component

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-3599)

### PR DESCRIPTION
**Pull request by nox1134**
Fixes #5368 
Please refer to the previous PR (https://github.com/mozilla/application-services/pull/5890). 
This new PR adds support for mockall for fxa_client.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by nox1134** [comment]
I did changes in this path only `components/fxa-client/` and all the test cases for this `fxa-client` also passed as well still while running tests for the whole repo, i am getting those errors.
`error: expect test failed
   --> components/remote_settings/src/client.rs:852:9` 
   You can check the CI to know about these errors. How do i fix these? and how are these related to changes that i did

**Review by tarikeshaq — CHANGES_REQUESTED** [review]
Thanks a ton @nox1134 🎉 🎉 

For the failing tests, I see there are quite a few changes to the `Cargo.lock` that would have caused it, you might have run `cargo update` at some point?

For now, if you revert the changes to `Cargo.lock` and run any cargo command, it'll be updated only to include the changes we'd expect (mockiato -> mockall) and CI should start passing from there! We'll fix the failing tests afterwards when we update our dependencies the next time - thank you so much 🙏

**Comment by nox1134** [comment]
Thanks for the review @tarikeshaq! Seems like it worked
How do i go about Check-formatting and other failed tests? I tried fixing the `clippy` tests but the issue was with the fact that whatever i fix in the `clippy` fails the cargo tests with `lifetime parameter` error.

File: components/fxa-client/src/internal/http_client.rs [hunk 1]
```diff
@@ -53,37 +53,37 @@ const DEVICES_FILTER_DAYS: u64 = 21;
 /// it return something called a "refresh token"? Using unambiguous
 /// verbs to start each method helps avoid confusion here.
 ///
-#[cfg_attr(test, mockiato::mockable)]
+#[cfg_attr(test, mockall::automock)]
```
Comment (tarikeshaq) [inline components/fxa-client/src/internal/http_client.rs:56]: Lets tell clippy to ignore the issue here, looks like we're hitting a limitation with mockall!
```suggestion
// Due to limitations in mockall, we have to add explicit lifetimes that the Rust compiler would have been happy to infer.
#[allow(clippy::needless_lifetimes)]
#[cfg_attr(test, mockall::automock)]
```

**Review by tarikeshaq — CHANGES_REQUESTED** [review]
Thanks again @nox1134! Lets ignore the clippy error in this case, and write a comment for our future selves why we did that (see my comment for what that looks like in practice)

for the formatting, try running `cargo fmt` then check-in the changes that the formatter makes

once we make those two changes we might be ready to land 👀 🎉

**Comment by tarikeshaq** [comment]
We're dealing with an intermittent test for the `ios-test` job, so you can ignore that failure (see #5902)

**Comment by tarikeshaq** [comment]
Awesome! This is almost ready to land, I ran the dependency check script (which attempts to make sure we include the licenses of any upstream dependencies) and there are some manual fixups that need to happen.. but not subject you to those fixups, I pushed a commit on top of your commits and will open it in a separate PR to land it (your commits are still there as-is to preserve authorship!)

File: components/fxa-client/Cargo.toml [hunk 2]
```diff
@@ -26,13 +26,14 @@ thiserror = "1.0"
 anyhow = "1.0"
 sync-guid = { path = "../support/guid", features = ["random"] }
 uniffi = "0.24.1"
+mockall = "0.11"
```
Comment (tarikeshaq) [inline components/fxa-client/Cargo.toml:29]: Aaah! this is why the dependency summaries were failing, lets remove this from there, it should only be a test dependency!

This way we don't need any fixups and we can land your PR without having to create a new one 👀

**Review by tarikeshaq — DISMISSED** [review]
Actually, I realized that the reason those fixups were needed was because we added `mockall` to the regular dependencies when it should've only lived as a test dependency!

If we fix this up we can land this PR without having to do any fixups to our scripts!

File: components/fxa-client/Cargo.toml (hunk 2 above)
Comment (nox1134) [reply components/fxa-client/Cargo.toml]: okay! thank you

**Review by tarikeshaq — APPROVED** [review]
Thank you so much @nox1134 🙏 I'll go ahead and land this

**Comment by tarikeshaq** [comment]
Aaah looks like one of the commits in your branch conflicts with the main branch (but not the full changset) - @nox1134 do you mind squashing your commits into a single commit with a descriptive name, then force-pushing? That way what lands is a clean single commit

**Comment by nox1134** [comment]
I squashed all my commits and force-pushed!
Big thanks 💯  @tarikeshaq for guiding me through contributing to Mozilla — I learned a lot while working on this issue. 🚀

**Review by tarikeshaq — APPROVED** [review]

**Comment by mhammond** [comment]
Hey @nox1134, thanks for your contribution and for sticking with this even after you hit a few hurdles!

**Pull request merged**